jcl/src/java.base/share/classes/java/lang/invoke/PermuteHandle.java

```diff
@@ -47,10 +47,19 @@
 	 */
 	@Override
 	MethodHandle permuteArguments(MethodType permuteType, int... permute2) {
+		if (isUnnecesssaryPermute(permuteType, permute)) {
```

Comment:
should `permute` be `permute2`?  I'd expect it to be using the parameter not the instance variable

Reply:
Thanks for catching this bug. Fixed it and another similar bug.